Tabbed document interface with AvalonDock

rvost opened the pull request:
I tried to prototype a tabbed GUI with [AvalonDock](https://github.com/Dirkster99/AvalonDock) instead of `Dragablz`. I haven't used this `AvalonDock` before (in DayzServerTools I'm currently use xceed's sibling fork of original project) but it looks promising and more customizable than `Dragablz`. 

The prototype isn't yet fully functional. I'm trying my best to keep it simple and not to push MVVM, however `TextEntry` and  `AboutEntry` are technically view models and project can benefit further from this approach. 

![image](https://user-images.githubusercontent.com/48875452/208319019-b8be7bb8-7bb8-4889-ae0c-40f55bf42fb8.png)

*(Theming example with VS2013 theme)*

So what do you think about it?

rvost committed:
feat(tabgui):  Scaffold tabbed document interface with AvalonDock

rvost committed:
feat(tabgui): Add save on close

rvost committed:
fix(tabgui): Non-functional fixes

- Recreate original UI layout;
- Remove redundant markup;
- Fix typo in TextEntry.

rvost committed:
fix(ux): Open new tabs on double click

- Open new tabs by double-clicking instead of selecting an entry to prevent accidentally opening too many tabs;
- Fix opening many tabs for one entry.

rvost committed:
feat(tabgui): Move focus to an open tab

Please note that implementing INotifyPropertyChanged on a window and binding to a window property is not normal practice and is for demonstration purposes only.

rvost committed:
fix(ux): Fix the focus problem when switching tabs using the keyboard

More info about the problem: https://www.codeproject.com/Articles/483533/AvalonDock-Tutorial-Part-Adding-a-Start-Page#premain544323

The TextEntry view has been moved to a separate user control as part of the solution.

FlipperPlz commented:
Sorry for the late response here and lack of commits this weekend! Was feeling abit sick but I'm finally feeling better. It would've been way better to notify you of this sooner but I plan on moving over to a MVVM architecture this is all new to me as I mostly make libraries and the only pattern I've used similar is MVC. Your work here looks great though, I took a quick look at it Saturday and another today. I will probably be reimplementing alot of it if I do go through with changing this project to reflect MVVM. Dont worry though as I'll be using this work for later commits

rvost committed:
refactor(tabgui): Scaffold view models for PBO entries

- Move TextEntry and AboutEntry to ViewModels namespace;
-  Add base class for PBO entries view model;

rvost commented:
I've cleaned a little bit structure of document view models. I think changes are mergeable now and there is no need to reimplement tabbed GUI part. Also it would be fairly straightforward to refactor `PboExplorerWindow` code-behind to MVVM and  integrate with this changes, I can help with that. Get well soon!

rvost commented:
Also there is couple of questions yet to be addressed: 
* exclude about page from ctrl-tab navigation,
* add context menu for tabs,
* possibly, save&load tabs layout on close.

Latter two should be simple to implement after introducing a view model for a tabbed interface (e.g. vm for whole `PboExplorerWindow`).

FlipperPlz commented:
Hello, I am back from my holiday break, sorry for the lack of replies. Hope you had a nice NewYears!

>I've cleaned a little bit structure of document view models. I think changes are mergeable now and there is no need to reimplement tabbed GUI part. Also it would be fairly straightforward to refactor PboExplorerWindow code-behind to MVVM and integrate with this changes, I can help with that. Get well soon!

Your changes and decisions look great judging off of the screenshots and UML provided in the issue, I should of merged this earlier when I commented but I completely forgot.

>Also there is couple of questions yet to be addressed:
 -    exclude about page from ctrl-tab navigation,
 -    add context menu for tabs,
 -    possibly, save&load tabs layout on close.
Latter two should be simple to implement after introducing a view model for a tabbed interface (e.g. vm for whole PboExplorerWindow).

I plan to have an entire preferences/options window that will allow a bunch of things to be changed, everything from pbo reading options (BisUtils allows you to specify how you want to read the PBO and what is/isn't allowed) to font size. This is obviously going to be later though once everything is put together


Thanks for this PR, I'll merge it now and start looking through it to make some edits!
